external/llvm-project/mlir/lib/Conversion/AMDGPUToROCDL/AMDGPUToROCDL.cpp

```diff
@@ -334,6 +340,57 @@ static Value mfmaConcatIfNeeded(ConversionPatternRewriter &rewriter,
   return input;
 }
 
+/// Push an input operand. If it is a float type, nothing to do. If it is
+/// an integer type, then we need to also push its signdness (1 for signed, 0
+/// for unsigned) and we need to pack the input 16xi8 vector into a 4xi32
+/// vector.
+static void wmmaPushInputOperand(ConversionPatternRewriter &rewriter,
+                                 Location loc, TypeConverter *typeConverter,
+                                 bool isUnsigned, Value llvmInput,
+                                 SmallVector<Value, 4> &operands) {
+  Type inputType = llvmInput.getType();
+  auto vectorType = inputType.dyn_cast<VectorType>();
+  Type elemType = vectorType.getElementType();
+
+  if (!elemType.isInteger(8)) {
+    operands.push_back(llvmInput);
+    return;
+  }
+
+  int64_t numBytes = vectorType.getNumElements();
+  Type i32 = rewriter.getI32Type();
+  VectorType vectorType32bits = VectorType::get(numBytes * 8 / 32, i32);
+  auto llvmVectorType32bits = typeConverter->convertType(vectorType32bits);
```

Comment:
... right, none of these are 4xi8, so vectors are fine. Never mind this comment I'm leaving for my own sanity